espnet2/spk/espnet_model.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2023 Jee-weon Jung
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+
+from contextlib import contextmanager
+from itertools import permutations
+from typing import Dict, Optional, Tuple
+
+import numpy as np
+import torch
+import torch.nn.functional as F
+from packaging.version import parse as V
+from typeguard import check_argument_types
+
+# TODO: remove unused modules
+from espnet2.asr.encoder.abs_encoder import AbsEncoder, RawNet3Encoder
+from espnet2.asr.frontend.abs_frontend import AbsFrontend
+from espnet2.asr.specaug.abs_specaug import AbsSpecAug
+from espnet2.diar.attractor.abs_attractor import AbsAttractor
+from espnet2.layers.abs_normalize import AbsNormalize
+from espnet2.torch_utils.device_funcs import force_gatherable
+from espnet2.train.abs_espnet_model import AbsESPnetModel
+from espnet.nets.pytorch_backend.nets_utils import to_device
+from espnet2.spk.loss import *
+
+if V(torch.__version__) >= V("1.6.0"):
+    from torch.cuda.amp import autocast
+else:
+    # Nothing to do if torch<1.6.0
```

Comment:
You don't need this check since the supported PyTorch versions in ESPnet are now all higher than 1.6.0.